### DISCUSSION TITLE
Fusilli v1.1.0

### DISCUSSION THREAD

**Discussion by florencejt** [florencejt/fusilli#15] [Announcements]
# Fusilli v1.1.0 Release Notes

It's time for a **FusilliFacelift<sup>TM</sup>**! 
There's been a couple changes this time around - but hopefully all to make the user experience that little bit better.

The most comprehensive change is that now you input your parameters into the data, training, and evaluation functions as is, instead of having to put them in a dictionary first.

So for example, instead of:
`params = {'batch_size': 8}`
`prepare_fusion_data(params)`

You would do:

`
prepare_fusion_data(batch_size=8)
`

This means that there are more arguments to input overall 😅 but it should be easier to see what Fusilli doing and easier to set up your experiments. 😍

## Major Changes
1. **Function Update**:
    - Renamed `get_data_module` to `prepare_fusion_data` for enhanced clarity.
2. **Parameter Handling**:
    - Fusilli now requires separate parameter input instead of a dictionary.
    - This boosts code transparency and simplifies bug tracking.
3. **Specifying input data paths**:
    - Input file paths should be put in a dictionary with keys "tabular1", "tabular2", and "image"
    - This is passed into `prepare_fusion_data`
4. **Handling External Data**:
    - To incorporate external data, create a dictionary of file paths akin to the input testing data.
    - Pass these paths into the `.from_new_data` methods within the evaluation figure classes.
5. **Directory Management**:
    - Input/output directory paths are now organized within a dictionary.
    - Keys: "checkpoints", "figures", and “losses”.
    - Training loss figures will be saved in a subdirectory named “losses” within the user-specified figures directory.
6. **Column Name Requirements in Tabular Data**:
    - Tabular data must now contain columns named “ID” and “prediction_label” in each row (previously: "study_id" and "pred_label").

## Minor Changes
1. **MCVAE Early Stopping**:
    - Adjust early stopping criteria in MCVAE using keyword arguments "mcvae_patience" and "mcvae_tolerance" passed to `prepare_fusion_data`.
2. **Weights and Biases Logging**:
    - Enable logging with Weights and Biases using the "wandb_logging" argument in `train_and_save_models`.

## Documentation Update
- Added an example notebook illustrating how to integrate external data with Fusilli.

For comprehensive guidance and examples, please refer to the updated documentation! 📖✨

<hr /><em>This discussion was created from the release <a href='https://github.com/florencejt/fusilli/releases/tag/v1.1.0'>Fusilli v1.1.0</a>.</em>
